### PR TITLE
Astro 1112 create login content

### PR DESCRIPTION
**Pull request by Design-Kat**
Updated Sign in content per @amalyah 's review on PR #147 

https://rocketcom.atlassian.net/browse/ASTRO-1112

### CODE REVIEW

**Comment by Design-Kat** [comment]
Ready for review by @dmcalester @amalyah @Cheshire89

File: _content/components/sign-in.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+tags: components
+path: /components/sign-in
+date: Last Modified
+layout: components.template.njk
+title: Sign In
+demo: components-buttons--all-button-variants
+storybook: components-buttons--standard-button
+git: rux-button
```
Comment (amalyah) [inline _content/components/sign-in.md:9]: @Design-Kat Will these aspects be changed by devs later?

File: _content/components/sign-in.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+The sign in page is presented to users in the following scenarios:
+When a user wants to gain access to an application.
+When a user has signed out voluntarily. They will see a confirmation message after which they will be automatically redirected back to the sign in page.
+When a user has been signed out due to inactivity. In this scenario, when the user signs back in they should be redirected to the last page they were on, before being signed out.
+
+## Rules of Thumb
+
+-  Use “Sign in”, not Login, Log in or Log-in. (Note: When leaving an application, use “Sign out”, not Sign off)
+-  Use a direct, actionable button name, such as Sign In, rather than a more generic “Submit” to post the form.
```
Comment (amalyah) [inline _content/components/sign-in.md:29]: Sign In > Sign in (sentence case rule)

File: _content/components/sign-in.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+-  Consult with your IT/Security team to ensure you are meeting your specific authentication best practices and security needs.
+-  Provide a way for users to sign out and re-authenticate.
+
+## Appearance and Behavior
+
+There are four established types of login workflows.
```
Comment (amalyah) [inline _content/components/sign-in.md:35]: login workflows > sign in workflows

File: _content/components/sign-in.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+Sign in with Username or Email Plus Password
+This is the most basic sign in workflow, wherein the user identifies and authenticates from the same screen using an ID and passcode.
+1. Sign in with Common Access Card (CAC)
+This sign in requires additional authentication via a CAC reader and PIN.
+2. Sign in with Single Sign-on (SSO)
+SSO allows the user to authenticate to multiple systems using one identification and passcode method.
+3. Login with Multi-Factor Authentication (MFA)
```
Comment (amalyah) [inline _content/components/sign-in.md:42]: Login > Sign in

File: _content/components/sign-in.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+MFA requires the user to set up additional methods of identification, such as a PIN, or
+4. The Sign in can currently be configured to display Sign in with Username or Email Plus Password, and validation.
```
Comment (amalyah) [inline _content/components/sign-in.md:44]: Sign in with Username > "Sign in with username"
Email Plus Password > "Email plus password"
(sentence case rule)

File: _content/components/sign-in.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+### Sign in with Username or Email Plus Password
+
+The sign in component consists of a field to identify the user, typically a username or email, and a masked field for a unique passcode, as well as a button that submits the content.  
+
+### Placement
+
+The login UI elements can be added in 3 possible placement areas.
```
Comment (amalyah) [inline _content/components/sign-in.md:52]: login UI > sign in UI

File: _content/components/sign-in.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+1. Modal dialog
+* Draws user attention to login process
```
Comment (amalyah) [inline _content/components/sign-in.md:55]: login process > sign in process

File: _content/components/sign-in.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+* Can allow application to be running in the background when signed out
+* Ideal for switching applications
+
+2. Sidebar
+* Generally placed on left side of page
+* Allows imagery/branding on opposite side of page to differentiate application
+* Often used alongside SSO
+
+3. Full page
+* Draws user attention to login process, without the need for a modal dialog to be placed over an essentially blank page
```
Comment (amalyah) [inline _content/components/sign-in.md:65]: login process > sign in process

File: _content/components/sign-in.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+* Ideal for initial entry points and when the user does not have access until after authentication
+
+
+## Validation
+
+Effective error messaging is important for creating great experiences. Not being able to log into an application is frustrating and blocks users from accomplishing their tasks.
```
Comment (amalyah) [inline _content/components/sign-in.md:71]: log into > sign into

File: _content/components/sign-in.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+Error messages should be clear and concise. They should help users understand what went wrong and give users steps to resolve the error. Be as specific as possible in your error messages
+Validate as much of the user’s data before submission as possible. This real-time validation should happen when the input field loses focus and checks for input errors like invalid characters and empty fields. This helps users easily identify mistakes and fix them before submitting the login form.
```
Comment (amalyah) [inline _content/components/sign-in.md:73]: login form > sign in form

File: _content/components/sign-in.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+Always present error states on the login screen, and use inline errors whenever possible when they do not present a security concern.
```
Comment (amalyah) [inline _content/components/sign-in.md:74]: login screen > sign in screen

File: _content/components/sign-in.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+Common client-side errors:
+* Empty required fields
+* Invalid characters
+* Incorrect input format
+
+If there are server-side errors when the user submits the login form, the page should be reloaded, the password field cleared, and the user returned to the username input field. Use an inline notification to display field-specific errors (where they do not provide a security concern) and provide clear direction on how users should resolve the issue, and a validation summary for errors that are not field-specific (such as time-outs, a server being down, or lack of connection) or for errors that apply to multiple fields and could present a security concern when inline (e.g. when the username is not found, or the password is correct they are often grouped into one error, such as “Username or password is incorrect”.)
```
Comment (amalyah) [inline _content/components/sign-in.md:81]: login form > sign in form

File: _content/components/sign-in.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+Common server-side errors:
+* Invalid authentication
+* Max # of login attempts reached
```
Comment (amalyah) [inline _content/components/sign-in.md:85]: login attempts > sign in attempts

File: _content/components/sign-in.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+* Account lock out
+* Database or system error
+
+## Accessibility
+
+Ensure that users can tab through the login form and navigate the page using only a keyboard. Use landmark regions to designate the login region and allow screen readers to skip directly to the input fields. This is especially important if you are using the split-screen layout or have additional content on the page.
```
Comment (amalyah) [inline _content/components/sign-in.md:91]: login form > sign in form
login region > sign in region

File: _content/components/sign-in.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+## Examples
+
+:::dos-donts
+![Do: Mask sensitive data, such as passwords](/img/components/component-template-do-1.png "Do: Something")
```
Comment (amalyah) [inline _content/components/sign-in.md:96]: I suspect the "Do: Something" pieces of the code in the dos and donts are supposed to be edited somehow.

**Review by amalyah — CHANGES_REQUESTED** [review]
There are some instances of log in/login that need to be changed to sign in.

File: _content/components/sign-in.md (hunk 15 above)
Comment (Design-Kat) [reply _content/components/sign-in.md:96]: Ouch. Good catch.

File: _content/components/sign-in.md (hunk 1 above)
Comment (Design-Kat) [reply _content/components/sign-in.md]: Note: Just committed updates from your comments- I commented these out for now, will update per Duncan's response.

File: _content/components/sign-in.md [hunk 16]
```diff
@@ -68,59 +68,56 @@ The login UI elements can be added in 3 possible placement areas.
...
+
+![Don't: Display sensitive data, such as passwords, unmasked by default](/img/components/replaceimage.png "Don't: Display sensitive data, such as passwords, unmasked by default")
 
-![Don't: Display sensitive data, such as passwords, unmasked by default](/img/components/component-template-dont-1.png "Don't: Something")
-Don't: Display sensitive data, such as passwords, unmasked by default
 :::
 
-![Do: Use clear button labels that describe specific tasks like "Sign in" or "Update password"](/img/components/component-template-do-1.png "Do: Something")
-Do: Use clear button labels that describe specific tasks like "Sign in" or "Update password"
+![Do: Use clear button labels that describe specific tasks like "Sign in" or "Update password"](/img/components/replaceimage.png "Do: Do: Use clear button labels that describe specific tasks like "Sign in" or "Update password")
```
Comment (amalyah) [inline _content/components/sign-in.md:103]: Double "Do:" in this quote.
Quotes in text should be changed to single quotes I believe. '
Need to close end of the "Do: ... " statement with a double quote. "

File: _content/components/sign-in.md [hunk 17]
```diff
@@ -68,59 +68,56 @@ The login UI elements can be added in 3 possible placement areas.
...
+
+![Don't: Use vague button labels like "Submit" to guide users through authentication flows](/img/components/replaceimage.png "Don't: Use vague button labels like "Submit" to guide users through authentication flows")
```
Comment (amalyah) [inline _content/components/sign-in.md:105]: @Cheshire89 will putting "Submit" in double quotes in this section cause problems for the code?

**Review by amalyah — CHANGES_REQUESTED** [review]
Small changes

File: _content/components/sign-in.md (hunk 17 above)
Comment (Design-Kat) [reply _content/components/sign-in.md:105]: @amalyah Changed all the double quotes within double quotes to single quotes, will update if need be based on @cheshire89's reply

File: _content/components/sign-in.md [hunk 18]
```diff
@@ -0,0 +1,123 @@
+---
+tags: components
+path: /components/sign-in
+date: Last Modified
+layout: components.template.njk
+title: Sign In
+[//]: #  demo: components-signin--all-signin-variants
+[//]: #  storybook: components-signin--standard-signin
+[//]: # git: rux-signin
```
Comment (cskerbo) [inline _content/components/sign-in.md:9]: This is generating a build error as these 3 mapping keys of `[//]` are considered duplicates and need to be unique.

**Review by cskerbo — CHANGES_REQUESTED** [review]

File: _content/components/sign-in.md (hunk 18 above)
Comment (Design-Kat) [reply _content/components/sign-in.md:9]: Ah, sorry - I was trying to comment out the code until those pages were completed and could be hooked up, and that is what the internet told me was how to do so in markup. Google fail :) 
Is there something else I can replace it with?

Comment (cskerbo) [reply _content/components/sign-in.md:9]: You should be able to just delete `[//]:` on each line. If you notice, the `#` successfully commented out the rest of the line.

```
# demo: components-signin--all-signin-variants
# storybook: components-signin--standard-signin
# git: rux-signin
```

Comment (cskerbo) [reply _content/components/sign-in.md:9]: Deleted my previous statement since we're dealing with Github markdown, you can do this to comment it out:

Comment (Design-Kat) [reply _content/components/sign-in.md]: Committed the update, thank you!

**Review by cskerbo — APPROVED** [review]

**Review by amalyah — APPROVED** [review]

**Review by dmcalester — APPROVED** [review]

**Pull request merged**